resource-managers/kubernetes/core/src/main/scala/org/apache/spark/scheduler/cluster/k8s/KubernetesClusterManager.scala

```diff
@@ -70,8 +70,16 @@ private[spark] class KubernetesClusterManager extends ExternalClusterManager wit
     // If/when feature steps are executed in client mode, they should instead take care of this,
     // and this code should be removed.
     if (!sc.conf.contains(KUBERNETES_EXECUTOR_POD_NAME_PREFIX)) {
-      sc.conf.set(KUBERNETES_EXECUTOR_POD_NAME_PREFIX,
-        KubernetesConf.getResourceNamePrefix(sc.conf.get("spark.app.name")))
+      val podNamePrefix = KubernetesConf.getResourceNamePrefix(sc.conf.get("spark.app.name"))
+      if (org.apache.spark.deploy.k8s.Config.isValidExecutorPodNamePrefix(podNamePrefix)) {
```

Comment:
Could you use `import` statement?

Reply:
ah, this is because the current import has an existed `io.fabric8.kubernetes.client.Config` so I use full name for the Spark `Config` here. But after we moved into `KubernetesUtils`, I think here is no this issue.